api/src/main/java/org/ccci/idm/user/Dn.java

```diff
@@ -0,0 +1,120 @@
+package org.ccci.idm.user;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+@Immutable
+public class Dn implements Serializable {
+    private static final long serialVersionUID = 5510344429904560934L;
+
+    private final List<Component> components;
+
+    public Dn() {
+        components = ImmutableList.of();
+    }
+
+    public Dn(@Nonnull final Component... components) {
+        this.components = ImmutableList.copyOf(components);
+    }
+
+    public Dn(@Nonnull final List<Component> components) {
+        this.components = ImmutableList.copyOf(components);
+    }
+
+    @Nonnull
+    public final List<Component> getComponents() {
+        return components;
+    }
+
+    @Nullable
+    public final String getName() {
+        return components.isEmpty() ? null : components.get(components.size() - 1).value;
+    }
+
+    public final boolean isDescendantOf(@Nonnull final Dn ancestor) {
+        return ancestor.components.size() < components.size() && Objects.deepEquals(ancestor.components, components.subList(0, ancestor
+                .components.size()));
+    }
+
+    public final boolean isAncestorOf(@Nonnull final Dn descendant) {
+        return descendant.isDescendantOf(this);
+    }
+
+    @Nonnull
+    public final Dn descendant(@Nonnull final Component... components) {
+        return new Dn(ImmutableList.<Component>builder().addAll(this.components).add(components).build());
+    }
+
+    @Nonnull
+    public final Group asGroup() {
```

Comment:
so, little bit extra code here, but simpler to understand logic at the application layer.